packages/samples/headless-react/src/components/cart-recommendations-list/cart-recommendations-list.class.tsx

```diff
@@ -0,0 +1,135 @@
+    this.setState(this.controller.state);
+  }
+
+  render() {
+    if (this.state.error) {
+      return (
+        <div>
+          <div>Oops {this.state.error.message}</div>
+          <code>{JSON.stringify(this.state.error)}</code>
+          <button onClick={() => this.controller.refresh()}>Try again</button>
+        </div>
+      );
+    }
+
+    if (!this.state.recommendations.length) {
+      return <button onClick={() => this.controller.refresh()}>Refresh</button>;
+    }
+
+    return (
+      <div>
+        <button onClick={() => this.controller.refresh()}>Refresh</button>
+        Based on your cart, you may like the following:
+        <ul style={{textAlign: 'left'}}>
+          {this.state.recommendations.map((product) => (
+            <li key={product.sku}>
+              <article>
+                <h2>
+                  <a href={product.link}>{product.name}</a>
+                </h2>
+                <img
+                  src={product.thumbnailUrl}
+                  alt={`Thumbnail of ${product.name}`}
+                  style={{opacity: product.inStock ? 1 : 0.5}}
+                />
+                {product.promoPrice !== product.price ? (
+                  <span>
+                    <s>{product.price}</s> <strong>{product.promoPrice}</strong>
+                  </span>
+                ) : (
+                  <strong>{product.price}</strong>
+                )}
+                <details>
+                  <summary>More details</summary>
+                  <table>
+                    <tr>
+                      <th>Rating</th>
+                      <td>{Math.round(product.rating)}/10</td>
+                    </tr>
+                    <tr>
+                      <th>Brand</th>
+                      <td>{product.brand}</td>
+                    </tr>
+                    <tr>
+                      <th>In stock</th>
+                      <td>{product.inStock ? 'Yes' : 'No'}</td>
+                    </tr>
+                    <tr>
+                      <th>Categories</th>
+                      <td>
+                        <ul>
+                          {product.categories.map((category) => (
+                            <li key={category}>{category}</li>
+                          ))}
+                        </ul>
+                      </td>
+                    </tr>
+                    <tr>
+                      <th>Tags</th>
+                      <td>
+                        <ul>
+                          {product.tags.map((tag) => (
+                            <li key={tag}>{tag}</li>
+                          ))}
+                        </ul>
+                      </td>
+                    </tr>
+                    <tr>
+                      <th>Identifier</th>
+                      <td>{product.sku}</td>
+                    </tr>
+                  </table>
+                </details>
+              </article>
+            </li>
+          ))}
+        </ul>
+      </div>
```

Comment:
I don't think it's necessary to demo every single property on the `product`, but just one, maybe two things. This would shorten the sample and make it more impactful in my opinion.

Ultimately though, I will defer to @francoislg for these commerce controllers.

As a smaller detail, I would discourage using tables for layout, because they are not responsive, and I _think_ worse for accessibility. Tables should primarily be used to display tabular data.

While these samples are not going to be rendered, devs may copy-paste/use the code as a starting point. For this reason, I believe its important to set people on the right path.